File: lib/features/bills/data/fetched_bill.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'fetched_bill.freezed.dart';

@freezed
abstract class FetchedBill with _$FetchedBill {
  const factory FetchedBill({
    required String billerId,
    required String account,
    required String customerName,
    required int amountPaise,
    required DateTime dueDate,
    required String billNumber,
  }) = _FetchedBill;
}

File: app/src/main/java/com/billpayments/features/bills/data/FetchedBill.kt
package com.billpayments.features.bills.data

import java.time.LocalDate

data class FetchedBill(
    val billerId: String,
    val account: String,
    val customerName: String,
    val amountPaise: Long,
    val dueDate: LocalDate,
    val billNumber: String,
)
